### ISSUE TITLE
linter: `--fix` removes only every other adjacent unused import, needs N passes to converge

### ISSUE THREAD

**Issue by senzujuju** [oxc-project/oxc#26943]
### What version of Oxlint are you using?

1.83.0

### What command did you run?

npx oxlint@1.83.0 repro.ts --fix

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "rules": {
    "no-unused-vars": ["error", { "fix": { "imports": "safe-fix" } }]
  }
}
```

### What happened?

A single `oxlint --fix` run removes only every *other* unused import when the
import statements are adjacent. Converging requires repeated invocations.

### Reproduction

No dependencies and no install needed — the imported modules do not have to exist.

`repro.ts`:

```ts
import { a } from './a';
import { b } from './b';
import { c } from './c';
import { d } from './d';
import { e } from './e';

export const keep = 1;
```

Run `npx oxlint@1.83.0 repro.ts --fix` once. Result:

```ts
import { b } from './b';
import { d } from './d';

export const keep = 1;
```

Lines 1, 3 and 5 were removed; lines 2 and 4 survived and are reported:

```
repro.ts:2:10: error eslint(no-unused-vars): Identifier 'b' is imported but never used.
repro.ts:4:10: error eslint(no-unused-vars): Identifier 'd' is imported but never used.
```

Repeating the same command converges in three runs total:

| run | imports left |
| --- | ------------ |
| 1   | 2 (`b`, `d`) |
| 2   | 1 (`d`)      |
| 3   | 0            |

### Control experiment — adjacency is the trigger

The same imports **separated by blank lines** are all removed in a single pass:

```ts
import { a } from './a';

import { b } from './b';

import { c } from './c';

export const keep = 1;
```

→ one `--fix` run leaves no imports.

This is consistent with `Fixer` skipping overlapping edits: removing an import
statement appears to consume its trailing newline, so two neighbouring removals
have touching ranges and all but the first are dropped for that pass.

### Expected behaviour

A single `--fix` run converges.
